contracts/utils/FlashLiquidator.sol

```diff
@@ -0,0 +1,154 @@
+import {IEulerMarkets} from "../../interfaces/euler/IEulerMarkets.sol";
+import {IEulerFlashLoanReceiver} from "../../interfaces/euler/IEulerFlashLoanReceiver.sol";
+import {NotionalProxy} from "../../interfaces/notional/NotionalProxy.sol";
+import {CErc20Interface} from "../../interfaces/compound/CErc20Interface.sol";
+import {CEtherInterface} from "../../interfaces/compound/CEtherInterface.sol";
+import {WETH9} from "../../interfaces/WETH9.sol";
+import {TokenUtils, IERC20} from "../utils/TokenUtils.sol";
+import {Constants} from "../global/Constants.sol";
+import {Token} from "../global/Types.sol";
+import {BoringOwnable} from "./BoringOwnable.sol";
+import {Deployments} from "../global/Deployments.sol";
+
+contract FlashLiquidator is IEulerFlashLoanReceiver, BoringOwnable {
+    using TokenUtils for IERC20;
+
+    NotionalProxy public immutable NOTIONAL;
+    address public immutable EULER;
+    IEulerMarkets public immutable MARKETS;
+    mapping(address => address) internal underlyingToAsset;
+
+    struct LiquidationParams {
+        uint16 currencyId;
+        address account;
+        address vault;
+        bytes redeemData;
+    }
+
+    constructor(NotionalProxy notional_, address euler_, IEulerMarkets markets_) {
+        NOTIONAL = notional_;
+        EULER = euler_;
+        MARKETS = markets_;
+        owner = msg.sender;
+        emit OwnershipTransferred(address(0), owner);
+    }
+
+    function enableCurrencies(uint16[] calldata currencies) external onlyOwner {
+        for (uint256 i; i < currencies.length; i++) {
+            (Token memory assetToken, Token memory underlyingToken) = NOTIONAL.getCurrency(currencies[i]);
+            IERC20(assetToken.tokenAddress).checkApprove(address(NOTIONAL), type(uint256).max);
+            if (underlyingToken.tokenAddress == Constants.ETH_ADDRESS) {
+                IERC20(address(Deployments.WETH)).checkApprove(address(EULER), type(uint256).max);
+                underlyingToAsset[address(Deployments.WETH)] = assetToken.tokenAddress;
+            } else {
+                IERC20(underlyingToken.tokenAddress).checkApprove(address(EULER), type(uint256).max);
+                IERC20(underlyingToken.tokenAddress).checkApprove(assetToken.tokenAddress, type(uint256).max);
+                underlyingToAsset[underlyingToken.tokenAddress] = assetToken.tokenAddress;
+            }
+        }
+    }
+
+    function estimateProfit(
+        address asset,
+        uint256 amount,
+        LiquidationParams calldata params
+    ) external onlyOwner returns (uint256) {
+        uint256 balance = IERC20(asset).balanceOf(address(this));
+        IEulerDToken dToken = IEulerDToken(MARKETS.underlyingToDToken(asset));
+        dToken.flashLoan(amount, abi.encode(asset, amount, false, params));
+        return IERC20(asset).balanceOf(address(this)) - balance;
+    }
+
+    function flashLiquidate(
+        address asset,
+        uint256 amount,
+        LiquidationParams calldata params
+    ) external {
```

Comment:
Is there any reason why we don't just combine this method with the estimate method and just have flashLiquidate return the diff in balances as well? Anyone can do a static call against these methods.